text/XXXX-externalize_private_keys.md

```diff
@@ -0,0 +1,111 @@
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Separate Qri user private key from configuration, storing in the operating system's keyring, or superceeded by an envrionrment variable `QRI_PRIVATE_KEY`.
+
+# Motivation
+[motivation]: #motivation
+
+Qri users are getting themselves into un-recoverable states by manually deleting their `.qri` directory, which takes their private key with it, creating major problems when they try to re-add & change datasets. Because Qri is built on public-key infrastructure (PKI), deleting the private key has the effect of locking a user out of their own identity.
+
+More importantly, storing private keys in configuration is a security vulnerability. Coupling sensitive information with configuration renders the entire configuration sensitive.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We inherited this bad habit of storing private keys in configuration as a "deal with it later" approach taken from IPFS, unlike Qri, IPFS identities are not "significant" in the sense that it's not tied to an identity intended for humans. Tying an identity to a private key means we need to deal with this problem much earlier.
+
+This RFC deprecates `profile.privKey` config value, moving the the private key to the user's operating system keyring. When no keyring is available, store the keyring in a seperate file called `QRI_PRIVATE_KEY` within the `.qri` directory, and present the user with a warning that they should copy this file to a safe location.
+
+In all cases the active private key can be overridden with an environment variable: `QRI_PRIVATE_KEY`. In all cases the key is expected to be a base64-encoded string.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Doing the work of generalizing secret storage can be handled by an external package: https://github.com/zalando/go-keyring. go-keyring supports Windos, OS X, and Linux. The key is stored in a `(username,password)` combination, with the "username" being the user's ProfileID (hash of public key), and the password being the base64-encoded private key. OS Keyrings support namespacing, qri's namespace will be a reversed-domain-name constant:
+
+```go
+const qriKeyringServiceID = "io.qri.keys"
+```
+
+Once setup is complete, `config.Profile.ID` stores the ProfileID string this configuration is attached to. Starting a qri process uses this as the "username" in a `Get` call:
+
+```go
+privKey, _ := keyring.Get(qriKeyringServiceID, config.Profile.ID)
+```
+
+### Keys subsystem
+
+All of this work should happen in a new package within the qri codebase: `github.com/qri-io/qri/keys`. We need to takle key rotation next, followed by storing _many_ keys for doing access control. All of this work should happen in a new subsystem called `keys.Service`. `lib.Instance` gets a new field that points to this service.
```

Comment:
Thoughts on putting this package in `github.com/qri-io/qri/identity/keys` or `github.com/qri-io/qri/config/keys`? I worry about having too many small top-level packages.